addons/stock_picking_wave/models/stock_picking_wave.py

```diff
@@ -1,77 +1,66 @@
 # -*- coding: utf-8 -*-
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
-from openerp.exceptions import UserError
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-class stock_picking_wave(osv.osv):
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class StockPickingWave(models.Model):
     _inherit = "mail.thread"
     _name = "stock.picking.wave"
     _description = "Picking Wave"
     _order = "name desc"
-    _columns = {
-        'name': fields.char('Picking Wave Name', required=True, help='Name of the picking wave', copy=False),
-        'user_id': fields.many2one('res.users', 'Responsible', track_visibility='onchange', help='Person responsible for this wave'),
-        'picking_ids': fields.one2many('stock.picking', 'wave_id', 'Pickings', help='List of picking associated to this wave'),
-        'state': fields.selection([('draft', 'Draft'), ('in_progress', 'Running'), ('done', 'Done'), ('cancel', 'Cancelled')], string="State", track_visibility='onchange', required=True, copy=False),
-    }
 
-    _defaults = {
-        'name': '/',
-        'state': 'draft',
-        }
+    name = fields.Char(string='Picking Wave Name', default='/', required=True, help='Name of the picking wave', copy=False)
+    user_id = fields.Many2one('res.users', string='Responsible', track_visibility='onchange', help='Person responsible for this wave')
+    picking_ids = fields.One2many('stock.picking', 'wave_id', string='Pickings', help='List of picking associated to this wave')
+    state = fields.Selection([('draft', 'Draft'), ('in_progress', 'Running'), ('done', 'Done'), ('cancel', 'Cancelled')], default ='draft', track_visibility='onchange', required=True, copy=False)
 
-    def confirm_picking(self, cr, uid, ids, context=None):
-        picking_todo = self.pool.get('stock.picking').search(cr, uid, [('wave_id', 'in', ids)], context=context)
-        self.write(cr, uid, ids, {'state': 'in_progress'}, context=context)
-        return self.pool.get('stock.picking').action_assign(cr, uid, picking_todo, context=context)
+    @api.multi
+    def confirm_picking(self):
```

Comment:
This is potentially a big mistake. You changed the operation order as well as method return, which can lead to regressions. The original method find pickings, update the waves, assign the pickings and return the assignation result. You do not respect that. Please update the method.